ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/util/VMUtil.java

```diff
@@ -96,12 +101,31 @@ public static String getVMInstallName(IExecutionEnvironment ee) throws CoreExcep
 		IPath containerPath = JavaRuntime.newJREContainerPath(ee);
 		IVMInstall vmi = JavaRuntime.getVMInstall(containerPath);
 		if (vmi == null) {
-			throw new CoreException(createErrorStatus(NLS.bind(UtilMessages.VMHelper_noJreForExecEnv, ee.getId())));
+			throw new CoreException(Status.error(NLS.bind(UtilMessages.VMHelper_noJreForExecEnv, ee.getId())));
 		}
 		return vmi.getName();
 	}
 
-	public static IStatus createErrorStatus(String message) {
-		return Status.error(message, null);
-	}
+	private static final Map<String, Double> JAVA_VERSION_OF_EE = Arrays.stream(getExecutionEnvironments())
+			.collect(Collectors.toUnmodifiableMap(IExecutionEnvironment::getId, ee -> {
+				return switch (ee.getId()) {
+				// The EE's handled explicitly have unexpected java targets.
+				// See https://github.com/eclipse-equinox/equinox/pull/655
+				case "J2SE-1.2" -> 1.2; //$NON-NLS-1$
+				case "J2SE-1.3" -> 1.3; //$NON-NLS-1$
+				case "J2SE-1.4" -> 1.4; //$NON-NLS-1$
+				default -> {
+					Properties properties = ee.getProfileProperties();
+					Object target = properties != null //
+							? properties.get(JavaCore.COMPILER_CODEGEN_TARGET_PLATFORM)
+							: null;
+					yield target instanceof String version ? Double.parseDouble(version) : 0;
+				}
```

Comment:
Just for clarification: This part could probably be simplified to only the default case if https://github.com/eclipse-equinox/equinox/pull/655 were available.